mirrord/intproxy/src/proxies/incoming/http.rs

```diff
@@ -1,86 +1,256 @@
-use std::convert::Infallible;
+use std::{fmt, io, net::SocketAddr};
 
-use bytes::Bytes;
-use http_body_util::combinators::BoxBody;
 use hyper::{
     body::Incoming,
     client::conn::{http1, http2},
-    Response, Version,
+    Request, Response, StatusCode, Version,
 };
 use hyper_util::rt::{TokioExecutor, TokioIo};
-use mirrord_protocol::tcp::HttpRequestFallback;
+use mirrord_protocol::{
+    tcp::{HttpRequest, HttpResponse, InternalHttpResponse},
+    ConnectionId, Port, RequestId,
+};
+use thiserror::Error;
 use tokio::net::TcpStream;
 use tracing::Level;
 
-use super::interceptor::{InterceptorError, InterceptorResult};
+mod client_store;
+mod response_mode;
+mod streaming_body;
 
-pub(super) const RETRY_ON_RESET_ATTEMPTS: u32 = 10;
+pub use client_store::ClientStore;
+pub use response_mode::ResponseMode;
+pub use streaming_body::StreamingBody;
 
-/// Handles the differences between hyper's HTTP/1 and HTTP/2 connections.
-pub enum HttpSender {
-    V1(http1::SendRequest<BoxBody<Bytes, Infallible>>),
-    V2(http2::SendRequest<BoxBody<Bytes, Infallible>>),
+/// An HTTP client used to pass requests to the user application.
+pub struct LocalHttpClient {
+    /// Established HTTP connection with the user application.
+    sender: HttpSender,
+    /// Address of the user application's HTTP server.
+    local_server_address: SocketAddr,
+    /// Address of this client's TCP socket.
+    self_address: SocketAddr,
```

Comment:
```suggestion
    address: SocketAddr,
```

This one time I think reducing the name here might be better, since it's kinda implied that this is kinda `self.address`, and `local_server_address` exists. Plus the name of the struct makes it hard to think of "what other address could this be".

Nit though.

Reply:
Changed